### PR TITLE
[f] 847 - posts by twitterbot should always include a in_reply_to_status_id parameter with the originating tweet ID

### PR DESCRIPTION
**Pull request by giang271291**
Description
--
* Add `in_reply_to` param to tweet API method

How to test
--
* Use `client.tweet` with `in_reply_to` param set to a tweet ID AND mention the tweets' owner's screen name in the reply text should result in a `conversation` on the replied tweet user timeline (a `View conversation` link is visible)

[#120698847] https://www.pivotaltracker.com/story/show/120698847

### CODE REVIEW

**Comment by steadykhoi** [comment]
LGTM

File: responsebot/responsebot_client.py [hunk 1]
```diff
@@ -44,14 +44,15 @@ def get_current_user(self):
...
+    def tweet(self, text, in_reply_to=None):
         """
         Post a new tweet.
 
         :param text: the text to post
+        :param in_reply_to: The ID of the tweet to reply to
         :return: Tweet object
         """
-        return Tweet(self._client.update_status(text)._json)
+        return Tweet(self._client.update_status(text, in_reply_to_status_id=in_reply_to)._json)
```
Comment (tklarryonline) [inline responsebot/responsebot_client.py]: I think when a keyword argument is used, other arguments should change to use kwargs too for consistency and better readability.

File: tests/unit_tests/test_tweet_client.py [hunk 2]
```diff
@@ -28,7 +28,17 @@ def test_post_new_tweet(self):
...
+        self.assertEqual(tweet.some_key, 'some value')
+
+    def test_reply(self):
+        self.real_client.update_status = MagicMock(return_value=MagicMock(_json={
+            'some_key': 'some value',
+        }))
+
+        tweet = self.client.tweet('some text', in_reply_to=123)
+
+        self.real_client.update_status.assert_called_once_with('some text', in_reply_to_status_id=123)
```
Comment (tklarryonline) [inline tests/unit_tests/test_tweet_client.py]: Same thought for the kwargs here

Comment (giang271291) [inline tests/unit_tests/test_tweet_client.py]: @tklarryonline ok will fix it, thanks dude

**Pull request merged**